src/bfuse32.rs

```diff
@@ -61,10 +65,8 @@ use bincode::{Decode, Encode};
 #[cfg_attr(feature = "bincode", derive(Encode, Decode))]
 #[derive(Debug, Clone)]
 pub struct BinaryFuse32 {
-    seed: u64,
-    segment_length: u32,
-    segment_length_mask: u32,
-    segment_count_length: u32,
+    #[cfg_attr(feature = "serde", serde(flatten))]
+    descriptor: Descriptor,
```

Comment:
I think this is technically a breaking change in bincode serialization/deserialization? Might have to flatten it back to what it was but let me know.